Makefile

```diff
@@ -266,6 +270,10 @@ build/src/libultra/libc/llcvt.o: src/libultra/libc/llcvt.c
 	python3 tools/set_o32abi_bit.py $@
 	@$(OBJDUMP) -d $@ > $(@:.o=.s)
 
+build/src/overlays/%_reloc.o: build/spec
+	$(FADO) $$(tools/reloc_preq $< $(notdir $(@D))) -n $(notdir $(@D)) -o $(@:.o=.s) -M $(@:.o=.d)
```

Comment:
If I understand this correctly, this assumes the directory of the `_reloc.o` include in the spec matches the overlay's segment name, which isn't great since the directory structure or overlay names might change in modding scenarios.
Ideally we should just extract the overlay name from the segment name in the spec, instead of looking at the `_reloc.o` path, but that might require quite a few changes so I'm not sure. Maybe we should just have a check for this and provide a clear error message to explain that the path must match the overlay name.

Reply:
Can confirm this still works.

Reply:
Using the filename itself isn't ideal but it's definitely better. I still feel like there should be a dedicated error check+message about the reloc filename not matching the overlay segment name, since it's a somewhat hidden requirement with this implementation, but that might just be me worrying too much about user error.

Reply:
Well, the `-n` argument of Fado makes it possible to use one with a different name, you'd just have to work out a sensible way to get it in there. I suppose I can add a check to `reloc_prereq` to do it.